barretenberg/cpp/src/barretenberg/stdlib/primitives/memory/rom_table.cpp

```diff
@@ -49,13 +52,19 @@ template <typename Builder> void rom_table<Builder>::initialize_table() const
         context->set_ROM_element(rom_id, i, entries[i].get_witness_index());
     }
 
+    // Preserve tags to restore them in the future lookups
```

Comment:
I think you forgot to initialize the tags inside the `rom_table` constructor

Reply:
It's a vector, so it is default-initialized